server/api/classyear/classyear.controller.js

```diff
@@ -141,17 +142,20 @@ exports.daycode = function(req, res){
         return res.send(200, classYear.dayCodes[i].code);
       }
     }
-    //Not ambigious code generator, function at the bottom.
-    var code = generateCode(6);
-
-    classYear.dayCodes.push({
-      date: today,
-      code: code,
-      bonusDay: req.body.bonusDay ? true : false
-    });
-    return classYear.save(function(err, classYear){
+    //unique code generator, function at the bottom.
+    uniqueDayCode(generateCode(6),6,function(err,dayCode){
```

Comment:
for testing, you can pass in any existing code for the first parameter. e.g uniqueDayCode(any-code , 6 , ......